pipeline/authn/authenticator_oauth2_introspection.go

```diff
@@ -168,8 +169,8 @@ func (a *AuthenticatorOAuth2Introspection) Authenticate(r *http.Request, session
 			return errors.WithStack(err)
 		}
 
-		if len(i.TokenType) > 0 && i.TokenType != "access_token" {
-			return errors.WithStack(helper.ErrForbidden.WithReason(fmt.Sprintf("Introspected token is not an access token but \"%s\"", i.TokenType)))
+		if len(i.TokenUse) > 0 && (i.TokenUse != "access_token" && i.TokenUse != "refresh_token") {
```

Comment:
Close :) We only want to allow `access_tokens` here as they are for access of APIs, and refresh_tokens are for refreshing access tokens :)

Reply:
:D
Got it and fixed 👯